hardware_interface/src/component_parser.cpp

```diff
@@ -231,6 +232,8 @@ HardwareInfo parse_resource_from_xml(const tinyxml2::XMLElement * ros2_control_i
       hardware.sensors.push_back(parse_component_from_xml(ros2_control_child_it) );
     } else if (!std::string(kTransmissionTag).compare(ros2_control_child_it->Name())) {
       hardware.transmissions.push_back(parse_component_from_xml(ros2_control_child_it) );
+    } else if (!std::string(kGPIOTag).compare(ros2_control_child_it->Name())) {
```

Comment:
Nit-pick: maybe move before transmission because is somehow logically related to `joint` and `sensor`